graphql-server/src/databases/database.resolver.ts

```diff
@@ -91,6 +116,29 @@ export class DatabaseResolver {
     return dbs;
   }
 
+  @Mutation(_returns => [String])
+  async databasesByConnection(
+    @Args() args: AddDatabaseConnectionArgs,
+  ): Promise<string[]> {
+    if (this.conn.getWorkbenchConfig()?.connectionUrl === args.connectionUrl) {
+      return this.databases();
+    }
+    const workbenchConfig = getWorkbenchConfigFromArgs(args);
+    const ds = getDataSource(workbenchConfig);
+
+    try {
+      await ds.initialize();
+      const { qf } = await newQueryFactory(workbenchConfig.type, ds);
+      const dbs = await qf.databases();
+      return dbs;
+    } catch (e) {
+      console.error("Error query databases:", e);
+      return [];
```

Comment:
You should remove this catch block we want this error to be surfaced. The graphql server will handle the thrown error and then hit the finally block